src/misc/materials.jl

```diff
@@ -0,0 +1,201 @@
+abstract type AbstractMaterials end
+
+using TOML, DocStringExtensions
+
+__abs_path_prefix__ = dirname(@__DIR__)
+MaterialProperties = TOML.parsefile(joinpath(__abs_path_prefix__,"material_data/MaterialProperties.toml"))
+
+"""
+$TYPEDEF
+
+Generic structural alloy.
+
+$TYPEDFIELDS
+"""
+struct StructuralAlloy <: AbstractMaterials
```

Comment:
I don't know if I would have done things in exactly this way, but I think this is good. It makes the individual materials into simple concrete structures. My solution would probably have been a bit more airy-fairy and abstract, with some sort of separate `MaterialProperties` structure for each material, but I like your way better.